File: src/main/java/net/frozenorb/potpvp/util/menu/menus/ConfirmMenu.java
package net.frozenorb.potpvp.util.menu.menus;

import net.frozenorb.potpvp.util.Callback;
import net.frozenorb.potpvp.util.menu.Menu;
import net.frozenorb.potpvp.util.menu.buttons.BooleanButton;
import lombok.AllArgsConstructor;
import lombok.Getter;
import net.frozenorb.potpvp.util.menu.Button;
import org.bukkit.Material;
import org.bukkit.entity.Player;

import java.util.HashMap;
import java.util.Map;

@AllArgsConstructor
public class ConfirmMenu extends Menu {

    private String title;
    @Getter private final Callback<Boolean> response;

    public Map<Integer, Button> getButtons(Player player) {
        HashMap<Integer, Button> buttons = new HashMap<>();

        for(int i = 0; i < 9; ++i) {
            if (i == 3) {
                buttons.put(i, new BooleanButton(true, this.response));
            } else if (i == 5) {
                buttons.put(i, new BooleanButton(false, this.response));
            } else {
                buttons.put(i, Button.placeholder(Material.STAINED_GLASS_PANE, (byte)14, " "));
            }
        }

        return buttons;
    }

    @Override
    public String getTitle(Player player) {
        return this.title;
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/menu/menus/ConfirmMenu.kt
package net.frozenorb.potpvp.kt.menu.menus

import net.frozenorb.potpvp.kt.menu.Button
import net.frozenorb.potpvp.kt.menu.Menu
import org.bukkit.entity.Player
import net.frozenorb.potpvp.kt.menu.buttons.BooleanButton
import org.bukkit.Material
import java.util.HashMap

class ConfirmMenu(private val title: String, private val callback: (Boolean) -> Unit) : Menu() {

    override fun getButtons(player: Player): Map<Int, Button> {
        val buttons = HashMap<Int, Button>()

        for (i in 0..8) {
            when (i) {
                3 -> buttons[i] = BooleanButton(true, callback)
                5 -> buttons[i] = BooleanButton(false, callback)
                else -> buttons[i] = Button.placeholder(Material.STAINED_GLASS_PANE, 14.toByte())
            }
        }

        return buttons
    }

    override fun getTitle(player: Player): String {
        return title
    }

}